servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -543,11 +636,11 @@ boolean addConnection(C connection) {
                     if (i == connections.length) {
                         break;
                     } else if (connections.length == 1) {
-                        if (currentConnState.state == State.ACTIVE) {
-                            if (connStateUpdater.compareAndSet(this, currentConnState, EMPTY_CONN_STATE)) {
+                        if (ActiveState.class.equals(currentConnState.state.getClass())) {
+                            if (connStateUpdater.compareAndSet(this, currentConnState, ACTIVE_EMPTY_CONN_STATE)) {
```

Comment:
Why closure of a connection resets the failure counter?

Reply:
Good catch. This logic was an optimization. Now It wouldn't hurt anyone as the trigger for counting is when there's 0 connections. But in the future if we make the trigger point configurable that could be a hidden bug.